beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/handler/GenericHandler.java

```diff
@@ -49,15 +49,17 @@
    * @param dryRunEnabled Dry Run boolean flag
    * @implNote This parent handler expects the child's cleanupPath call to update & remove the record from this call such
    * that subsequent DB queries will not return the record. Hence why we only call next during dryRuns where no updates occur.
-   * @implNote Note that we only expect pageable.next to be called on
+   * @implNote Note that we only expect pageable.next to be called during a dry run.
+   * @return Pageable to pass to query. In the case of dry runs, this is the next page.
    */
-  public void processPage(Pageable pageable, Page<EntityHousekeepingPath> page, boolean dryRunEnabled) {
+  public Pageable processPage(Pageable pageable, Page<EntityHousekeepingPath> page, boolean dryRunEnabled) {
```

Comment:
When this was changed to `void` in the other PR I remember thinking that it was strange and that we had returned a `Pageable` for a reason, so this is why!

Reply:
I should have asked the question.

Reply:
I completely missed it!

Reply:
Yeah looking into this a bit more I believe this is something that just fell through the cracks between some of the refactors & comments. 

For some additional context: I think at one point when I made these changes I had the intention of handing off all the paging aspects to processPage so that `PagingCleanupService#pagingCleanup. That functionality was reverted at some point but I missed some pieces of the revert apparently. ( sorry :( )

Reply:
Hey @KenFigueiredo absolutely no worries, we all missed it, just one of those things that slips through. Thanks for all your contributions!